### ISSUE TITLE
The 'mip' keyword in recipe yaml is confusing 

### ISSUE THREAD

**Issue by goord** [ESMValGroup/ESMValCore#834]
The 'mip' field in the datasets dictionaries refer to CMIP-tables, not to MIP's themselves. There is certainly not a one-to-one correspondence between the two concepts, so I suggest to rename the key to e.g. 'table'

**Comment by Peter9192** [comment]
> > This is only the case for model data. Note that we also support observational datasets.
> 
> The exact reason we replaced the `model` tag with `dataset`

**Comment by Peter9192** [comment]
Sorry, bad click. I wanted to suggest that if we use the CMIP6 DRS as reference, then we could also consider changing 'dataset' to 'source_id' or just plain 'source'. I think that may eventually be clearer, but we might not want to change this too often...

**Comment by zklaus** [comment]
I agree that `mip` is a bit of a confusing name[^1] and that `table` is a suitable replacement.

Basically, all the information about the structure of CMIP6 output is linked [here](https://pcmdi.llnl.gov/CMIP6/Guide/modelers.html#5-model-output-requirements), the global attributes are described in [Taylor et al. (2018)](http://goo.gl/v1drZl).

As for `dataset` and the general approach, I think we have chosen not to follow any single project too closely in nomenclature and that has served us well. It reduces the trouble of mapping all projects into our common nomenclature, but maybe we should have such a mapping clearly in the documentation. Then again, maybe we do? An example can be found [here](https://github.com/ESMValGroup/ESMValCore/pull/183#issuecomment-515070248).

Heads up: You probably know that the tables and their names are somehow a strange amalgamation of realms, frequency, and other information. This was deemed convenient when the effort started by sending around spreadsheet files with different tables on different sheets. Now there is talk of moving away from it and going to a clearer separation, perhaps doing away with tables altogether in favor of the use of the frequency and realm facets, perhaps in CMIP7.

[^1] MIP on its own now usually refers to the expert groups that define experiments and request variables, such as OMIP, CFMIP, C4MIP, ... have a look at [the CMIP6 endorsed MIPs](https://www.wcrp-climate.org/modelling-wgcm-mip-catalogue/modelling-wgcm-cmip6-endorsed-mips) and links therein if interested. Then of course there is CMIP, the Coupled Model Intercomparison Project, itself and, as a twist, CMIP, the CoreMIP, which is the MIP used for the DECK(+) runs.
